pkg/cri/opts/spec.go

```diff
@@ -51,27 +51,51 @@ func WithoutRoot(ctx context.Context, client oci.Client, c *containers.Container
 	return nil
 }
 
-// WithProcessArgs sets the process args on the spec based on the image and runtime config
-func WithProcessArgs(config *runtime.ContainerConfig, image *imagespec.ImageConfig) oci.SpecOpts {
-	return func(ctx context.Context, client oci.Client, c *containers.Container, s *runtimespec.Spec) (err error) {
-		command, args := config.GetCommand(), config.GetArgs()
-		// The following logic is migrated from https://github.com/moby/moby/blob/master/daemon/commit.go
-		// TODO(random-liu): Clearly define the commands overwrite behavior.
-		if len(command) == 0 {
-			// Copy array to avoid data race.
-			if len(args) == 0 {
-				args = append([]string{}, image.Cmd...)
+func getArgs(imgEntrypoint []string, imgCmd []string, ctrEntrypoint []string, ctrCmd []string) ([]string, bool, error) {
+	// firstArgFromImg is a flag that is returned to indicate that the first arg in the slice comes from either the image
+	// Entrypoint or Cmd. If the first arg instead comes from the container config (e.g. overriding the image values),
+	// it should be false.
+	// Essentially this means firstArgFromImg should be true iff:
+	// - Container entrypoint is nil and image entrypoint is not empty
+	// - Container entrypoint is nil and image entrypoint is empty and container cmd is empty and image cmd is not empty
+	// - Container entrypoint is empty and container cmd is empty and image cmd is not empty
```

Comment:
A table would help here (although hard to do in comments..), or commas to help break em up a little